dv/uvm/core_ibex/collect_results.py

```diff
@@ -4,78 +4,217 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import argparse
+import collections
+import junit_xml
+import os.path
 import sys
-from typing import TextIO
+import yaml
+from test_run_result import TestRunResult, test_run_result_fields, \
+    check_test_run_result
+from typing import List, TextIO
+
+
+def parse_test_run_result(path: str) -> TestRunResult:
+    try:
+        with open(path) as yaml_file:
+            test_run_result_dict = yaml.load(yaml_file, Loader=yaml.SafeLoader)
+            loaded_fields = test_run_result_dict.keys()
+            if set(loaded_fields) != set(test_run_result_fields):
+                raise RuntimeError(f'Error loading YAML at {path}: does not '
+                        'contain the correct set of fields')
+
+            trr = TestRunResult(**test_run_result_dict)
+            try:
+                check_test_run_result(trr)
+            except AssertionError as e:
+                raise RuntimeError(f'Error loading YAML at path {path}: ' \
+                    'field types were incorrect')
+            return trr
+    except IOError as e:
+        raise RuntimeError(f'Error loading YAML at path {path}: {e}')
+    except yaml.YAMLError as e:
+        raise RuntimeError(f'Error parsing YAML at path {path}: {e}')
```

Comment:
There's a neat syntax for catching two different types of errors:
```suggestion
    except (IOError, yaml.YAMLError) as e:
        raise RuntimeError(f'Error parsing YAML at path {path}: {e}')
```